pkg/config/analysis/analyzers/gateway/gateway.go

```diff
@@ -84,7 +84,11 @@ func (*IngressGatewayPortAnalyzer) analyzeGateway(r *resource.Instance, c analys
 				if svcSelector.Matches(podLabels) {
 					for _, port := range service.Ports {
 						if port.Protocol == "TCP" {
-							servicePorts[uint32(port.Port)] = true
+							if tp := port.TargetPort.IntValue(); tp != 0 {
```

Comment:
Should we update the descriptions above as well?

Reply:
https://github.com/istio/istio/blob/f336e74304772bcc91100d3c7775524b2b18448a/pkg/config/analysis/analyzers/gateway/gateway.go#L32-L49

some comments and also the analyzer description

Reply:
I believe it is necessary to provide a comment explaining why the choice of targetPort is made, as I have also had doubts about it in the past.